### DISCUSSION TITLE
How to solve the problem that there is only one column from BIRT export to xlsx

### DISCUSSION THREAD

**Discussion by 858882847** [eclipse-birt/birt#1240] [Q&A]
My BIRT has nested three levels of table exports to xlsx, and the contents are all in the first column of Excel. How to solve the problem of nested three levels of table exports to xlsx and xls in a BIRT
[zcbh.zip](https://github.com/eclipse/birt/files/11028049/zcbh.zip)

**Comment by wimjongman** [comment]
Did you make an effort to try any of this yourself?

**Reply by 858882847** [reply]
I have tried exporting xls or pdf with BIRT, which is what I want. However, exporting xlsx does not have the style I want, and I don't know how to make it look like xls or pdf.

**Comment by speckyspooky** [comment]
I verified your report again and there are many issues on it.
1. The first thing is you should be more write structured your scripts so that it is more readable. This is important for further maintenance and support on your side.

2. Please work with the steps to start at first the preview (HTML) and create there your excel because so you figure out many of your issues. It is always better to use the run & render task of BIRT.

3. Further issues:

A) Your assigment of "display = none" is wrong in this report version to because none is a string value so you need there the string quotes.

B) You use an variable "count1" which is never defined

C) You should add Java package import if you doesn't use this like your line "importPackage(Packages.java.lang);" because this has an impact of the BIRT-RHINO-engine. Remove the line.

D) I don't know why you have added an table arround the 3 tables if you need only a layout element a grid would be better.

E) For Excel the wrapped table/grid must be have the same column count of the table with the "maximum" columns. The emitter counts the first table structure an create based on it the column count for export and your 1th table has only 1 column.

F) Your inner tables have again a width on table level. This value will be added if you use the UI-table mover with the mouse instead of the property editor.

G) You have added all 3 tables in 1 cell therefore all tables will be created in excel side by side instead of sheets.

H) The merge of the cell-value with the "minus"-lines caused issues because the existing merges on your structure. Therefore you must avaoid there the merge of cells. perhaps you cann use a border line.

Please be aware you must be a little bit more structured on the script level and sometimes a little bit try-and-error will help to figure out issues by your self.
Attached you get a small improved version also with the excel but I haven't fixed all issue. It is your task because you know the requirements of your customer.

[zcbh_speckyspooky.zip](https://github.com/eclipse/birt/files/11033078/zcbh_speckyspooky.zip)

**Reply by 858882847** [reply]
![2V)IV@9DAEN66PP${GCYGL3](https://user-images.githubusercontent.com/117696273/226773491-9ef9060b-30e2-41eb-87e9-293c96c3006f.png)
Can xlsx export products like xls

**Reply by speckyspooky** [reply]
If you use Spudsoft XLS and Spudsoft XLSX then you will get the same result.
The old XLS-Excel output isn't supported and the content isn't real excel too it is the excel-xml file type which simulate the xls.

Your screen lloks good.

**Reply by 858882847** [reply]
It's a bit regretful that it would be great if the old XLS-Excel and the new XLSX-Excel could be combined. The old XLS does not support images and its underlying file type is not a true Excel file type; it is an XML file type. When exporting data with XLSX-Excel, one record is normal, but multiple records are paginated. Moreover, my BIRT report contains three nested tables, but the exported pagination separates the three tables, and I don't know how to solve this problem.

**Reply by 858882847** [reply]
![}2H57AR 35G@9} ( L44 YF](https://user-images.githubusercontent.com/117696273/226857695-6111fb04-a02a-43f6-bdeb-1776fcdd63ac.png)

If I set BIRT to export XLSX-Excel files in this way, it meets my requirements, but I feel like there might be problems. Is there a better way to do it?

**Reply by speckyspooky** [reply]
Yes, of course, like one of my posts before I told you that the old XLS is the excel-XML
and the new Spudsoft-excel-file-types (XLS & XLSX) are the new version based on "apache POI library".

And Spudsoft as vendor has scoped to create an excel emitter for data export and not for beautified perfect all in one excel-XLSX-file type.
If you get with your last screen the one pager then use it in this way.
Another option woud be an user property at report level with the name ExcelEmitter.SingleSheet -> Type: Boolean -> Value: True

The Spudsoft emitter will only create 1 sheet but you have to handle the repeated headlines which should be hidden.

**Reply by speckyspooky** [reply]
But I would say your original question is answered -> 1 column problem.
